Backpack/src/main/java/net/skyscanner/backpack/chip/BpkChip.kt

```diff
@@ -45,53 +45,72 @@ open class BpkChip @JvmOverloads constructor(
       this.isEnabled = !disabled
     }
 
-  private val style: BpkChipStyle
+  private var _style: BpkChipStyle
 
+  @Deprecated("Custom background colour is no longer supported. Use existing styles")
   var chipBackgroundColor: Int
-    get() = style.backgroundColor
+    get() = _style.backgroundColor
     set(value) {
-      style.backgroundColor = value
+      _style.backgroundColor = value
       updateStyle()
     }
 
+  @Deprecated("Custom text colour is no longer supported. Use existing styles")
   var chipTextColor: Int
-    get() = style.textColor
+    get() = _style.textColor
     set(value) {
-      style.textColor = value
+      _style.textColor = value
       updateStyle()
     }
 
+  @Deprecated("Custom background colour is no longer supported. Use existing styles")
   var selectedBackgroundColor: Int
-    get() = style.selectedBackgroundColor
+    get() = _style.selectedBackgroundColor
     set(value) {
-      style.selectedBackgroundColor = value
+      _style.selectedBackgroundColor = value
       updateStyle()
     }
 
+  @Deprecated("Custom background colour is no longer supported. Use existing styles")
   var disabledBackgroundColor: Int
-    get() = style.disabledBackgroundColor
+    get() = _style.disabledBackgroundColor
     set(value) {
-      style.disabledBackgroundColor = value
+      _style.disabledBackgroundColor = value
       updateStyle()
     }
 
+  var style: Style?
+    get() = _style.style
+    set(value) {
+      if (value != null) {
+        _style = BpkChipStyles.Solid.fromTheme(context, value)
+        updateStyle()
+      }
+    }
+
+  var type: Type = Type.Option
+    set(value) {
+      field = value
+      updateIcons()
+    }
+
   var icon: Drawable? = null
     set(value) {
       field = value
         ?.mutate()
-        ?.apply { setBounds(0, 0, iconSize, iconSize) }
-        ?.also {
-          it.setTintList(textColors)
-          this.setCompoundDrawablesRelative(it, null, null, null)
+        ?.apply {
+          setBounds(0, 0, iconSize, iconSize)
+          setTintList(textColors)
         }
+      updateIcons()
```

Comment:
we now have two icon options, so had to update setting the drawable to a separate function